contracts/CompoundV3InvestStrategy.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import {ICompoundV3} from "./interfaces/ICompoundV3.sol";
+import {ICometRewards} from "./interfaces/ICometRewards.sol";
+import {IERC20} from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import {IAccessControl} from "@openzeppelin/contracts/access/IAccessControl.sol";
+import {SwapLibrary} from "@ensuro/swaplibrary/contracts/SwapLibrary.sol";
+import {IInvestStrategy} from "./interfaces/IInvestStrategy.sol";
+import {StorageSlot} from "@openzeppelin/contracts/utils/StorageSlot.sol";
+import {IExposeStorage} from "./interfaces/IExposeStorage.sol";
+import {InvestStrategyClient} from "./InvestStrategyClient.sol";
+
+/**
+ * @title CompoundV3ERC4626
```

Comment:
copypasta from the other contract here. Also, both here and on CompountV3ERC4626, exiting the vault is not permissioned.